### PR TITLE
chore: code reorganization

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This change moves our source files under `src`, our build files under `build`, and uses the base `tsconfig-google.json` config.  It makes this library more consistent with our other libraries, and makes it easier to work with generated files.

### CODE REVIEW

**Comment by JustinBeckwith** [comment]
@alexander-fenster Would be great if you could start reviewing these :)

File: src/scripts/generator.ts [hunk 1]
```diff
@@ -222,7 +223,7 @@ export class Generator {
 
   generateIndex(callback: Function) {
     const apis = {};
-    const apisPath = path.join(__dirname, '../apis');
+    const apisPath = path.join(__dirname, '../../../src/apis');
```
Comment (alexander-fenster) [inline src/scripts/generator.ts]: move src/apis to constant as well?

Comment (JustinBeckwith) [reply src/scripts/generator.ts]: fixed!

**Comment by white0ut** [comment]
Please note that this change broke link refs for KMS: https://cloud.google.com/kms/docs/reference/libraries#client-libraries-install-nodejs

You may want to check with other products referencing here to make sure they aren't broken.

**Comment by JustinBeckwith** [comment]
Yar, thanks for the heads up.  @jmdobry is there an easy way to search the docs for links to this repo?

**Pull request merged**